src/dlite-errors.h

```diff
@@ -8,14 +8,14 @@ typedef enum {
   dliteUnknownError = -1,
   dliteIOError = -2,
   dliteRuntimeError = -3, /* originally -1 */
-  dliteIndexError = -4,
-  dliteTypeError = -5,
+  dliteIndexError = -4, // e.g. x = [1,2,3], x[7]
+  dliteTypeError = -5, // e.g. float / str
   dliteDivisionByZero = -6,
   dliteOverflowError = -7,
   dliteSyntaxError = -8,
-  dliteValueError = -9,
-  dliteSystemError = -10,
-  dliteAttributeError = -11,
+  dliteValueError = -9, // e.g. int('apple')
+  dliteSystemError = -10, 
+  dliteAttributeError = -11, // e.g. obj.b (if b not defined) x=10, x.append(6)
   dliteMemoryError = -12,
   dliteNullReferenceError = -13,
   dliteStorageOpenError = -14,     /*!< cannot open storage plugin */
```

Comment:
use this comment style so we can generate doxygen comments. Should probably expand documentation of error codes in general. See below for new error codes to be added

Reply:
> use this comment style so we can generate doxygen comments. Should probably expand documentation of error codes in general. See below for new error codes to be added

Use the comment style suggested [above](https://github.com/SINTEF/dlite/pull/351/files/5a5c5d7dc10f4a147af155297f3ddbaac818fbd6#r986462281)

Reply:
Please take a look at how I did it here